app/controllers/chat_channels_controller.rb

```diff
@@ -64,21 +64,19 @@ def unfollow
   def create
     params.require([:type, :id, :name])
     guardian.ensure_can_create_chat_channel!
-    raise Discourse::InvalidParameters unless ["topic", "category"].include?(params[:type].downcase)
+    raise Discourse::InvalidParameters unless params[:type].downcase == 'category'
     raise Discourse::InvalidParameters.new(:name) if params[:name].length > SiteSetting.max_topic_title_length
 
-    creating_topic_channel = params[:type].downcase === "topic"
-    chatable_type = creating_topic_channel ? "Topic" : "Category"
+    chatable_type = "Category"
     existing_args = {
       chatable_type: chatable_type,
       chatable_id: params[:id]
     }
-    existing_args[:name] = params[:name] unless creating_topic_channel
+    existing_args[:name] = params[:name]
```

Comment:
This line could be moved into the `existing_args` declaration right above now that there aren't any conditions